Remove use of `pkg_resources`

kbwestfall opened the pull request:
As titled.  `pkg_resources` is being deprecated, so this changes the relevant functions to use `importlib` instead.  Changes mostly touch the scripts used for automated doc construction, but it also touches the remote file download/caching system.  I'll run the dev-suite just to be sure.

kbwestfall committed:
remove use of pkg_resources

kbwestfall committed:
rm comment

kbwestfall committed:
changes modifications

kbwestfall committed:
python39 fix?

tbowers7 committed:
Going all-in on pathlib

In the `doc/scripts` directory, replace all possible `os` calls with
their equivalent `pathlib` call.  Also, changed string formatting to
f-strings where applicable.

In `pypeit.data.utils`, greatly simplified the Path class in addition
to shifting from `pkg_resources` to `importlib`.

	modified:   doc/conf.py
	modified:   doc/scripts/build_afterburn_datamodels_rst.py
	modified:   doc/scripts/build_bitmask_rst.py
	modified:   doc/scripts/build_calib_images_rst.py
	modified:   doc/scripts/build_datacontainer_datamodels.py
	modified:   doc/scripts/build_dependency_rst.py
	modified:   doc/scripts/build_detector_table.py
	modified:   doc/scripts/build_imgproc_defaults.py
	modified:   doc/scripts/build_par_rst.py
	modified:   doc/scripts/build_specobj_rst.py
	modified:   doc/scripts/build_spectbl_rst.py
	modified:   doc/scripts/make_example_files.py
	modified:   doc/scripts/unique_configuration_keys.py
	modified:   doc/scripts/write_script_help.py
	modified:   pypeit/data/utils.py

tbowers7 committed:
Merge pull request #1687 from tbowers7/pkgsrc_tbowers

Going all-in on `pathlib`

tbowers7 reviewed (approved):
Thanks for getting on top of the deprecated `pkg_resources` API.  Looks good to me.

profxj reviewed (approved):
nice work and much appreciated.

tbowers7 commented on doc/conf.py:
```diff
@@ -12,21 +12,20 @@
-conf.read([os.path.join(os.path.dirname(__file__), '..', 'setup.cfg')])
-setup_cfg = dict(conf.items('metadata'))
+conf.read([pypeit_root / "setup.cfg"])
+setup_cfg = dict(conf.items("metadata"))
```
@kbwestfall : Be sure this change is compatible with RTD... I used the same syntax on a package of my own (GH pages rather than RTD) and needed to revert to the `os.path...` version because `resources.files()` pulls the installed package location (somewhere in `site-packages`) rather than the source code location (which includes the desired `setup.cfg` file).

tbowers7 replied:
I am reverting this problematic change that I introduced.

tbowers7 committed:
Revert problematic change in

tbowers7 committed:
More reversions

tbowers7 commented:
@kbwestfall : The reversions in 4d89d3ead8f72b0995ce60b54f59cf53189e364b take these files back to before my #1687 PR.

tbowers7 committed:
Carefully just convert `os` -> `pathlib`

	modified:   doc/scripts/build_bitmask_rst.py
	modified:   doc/scripts/build_dependency_rst.py
	modified:   doc/scripts/build_detector_table.py
	modified:   doc/scripts/build_imgproc_defaults.py
	modified:   doc/scripts/build_par_rst.py
	modified:   doc/scripts/build_specobj_rst.py
	modified:   doc/scripts/build_spectbl_rst.py
	modified:   doc/scripts/make_example_files.py
	modified:   doc/scripts/write_script_help.py

tbowers7 committed:
One more

kbwestfall committed:
Merge branch 'develop' into pkgrsrc

kbwestfall committed:
minor fixes

kbwestfall commented:
Running the dev-suite again...

tbowers7 commented on doc/conf.py:
```diff
@@ -18,7 +18,7 @@
 # Get configuration information from setup.cfg
 conf = ConfigParser()
-conf.read([pathlib.Path(__file__).resolve.parents[1] / 'setup.cfg'])
+conf.read([pathlib.Path(__file__).resolve().parents[1] / 'setup.cfg'])
```
Whoops, sorry.  That was me.

tbowers7 commented on doc/scripts/write_script_help.py:
```diff
@@ -27,7 +27,7 @@ def write_help(script_cls, opath, width=80):
     pypeit_root = resources.files('pypeit').parent
     path = pypeit_root / 'doc' / 'help'
-    if path.is_dir():
+    if not path.is_dir():
```
Whoops, sorry.  That was me.
